File: 1. Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        n = len(nums)
        hash_res = {}
        for idx, val in enumerate(nums):
            req = target - val
            if req in hash_res:
                return [idx, hash_res[req]]
            else:
                hash_res[val] = idx
        

File: 1. Two Sum.go
func twoSum(nums []int, target int) []int {
    res_hash := make(map[int]int)
    for i := range(len(nums)){
        req := target - nums[i]
        if _, ok := res_hash[req]; ok{
            return []int{res_hash[req], i}
        }else {
            res_hash[nums[i]] = i
        }
    }
    return []int{}
}